roachtest: import/tpch/nodes=8 failed

SHA: https://github.com/cockroachdb/cockroach/commits/b8554ec29fd1620c0e6af9544d678db57d251f4c

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=import/tpch/nodes=8 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1406588&tab=buildLog

```
The test failed on branch=release-19.1, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20190726-1406588/import/tpch/nodes=8/run_1
	cluster.go:2090,import.go:188,test_runner.go:691: import failed: pq: internal error: uncaught error: read tcp 10.128.0.115:42360->172.217.212.128:443: read: connection reset by peer
		row 14092578: reading CSV record
		github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*csvInputReader).readFile
			/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_csv.go:132
		github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*csvInputReader).readFile-fm
			/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_csv.go:80
		github.com/cockroachdb/cockroach/pkg/ccl/importccl.readInputFiles.func1
			/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_proc.go:140
		github.com/cockroachdb/cockroach/pkg/ccl/importccl.readInputFiles
			/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_proc.go:149
		github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*csvInputReader).readFiles
			/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_csv.go:80
		github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*readImportDataProcessor).doRun.func1
			/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_proc.go:488
		github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1
			/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:170
		github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
			/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		runtime.goexit
			/usr/local/go/src/runtime/asm_amd64.s:1333
		gs://cockroach-fixtures/tpch-csv/sf-100/lineitem.tbl.1
		github.com/cockroachdb/cockroach/pkg/ccl/importccl.readInputFiles.func1
			/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_proc.go:141
		github.com/cockroachdb/cockroach/pkg/ccl/importccl.readInputFiles
			/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_proc.go:149
		github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*csvInputReader).readFiles
			/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_csv.go:80
		github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*readImportDataProcessor).doRun.func1
			/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_proc.go:488
		github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1
			/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:170
		github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
			/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		runtime.goexit
			/usr/local/go/src/runtime/asm_amd64.s:1333

```

[(roachtest).import/tpch/nodes=8 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1676715&tab=buildLog) on [master@52261c39c65aecf960e3c8d54089a66d378af610](https://github.com/cockroachdb/cockroach/commits/52261c39c65aecf960e3c8d54089a66d378af610):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20200107-1676715/import/tpch/nodes=8/run_1
	cluster.go:2189,import.go:164,test_runner.go:716: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2180
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2188
		    main.registerImportTPCH.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/import.go:164
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:716
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2233
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - pq: gs://cockroach-fixtures/tpch-csv/sf-100/lineitem.tbl.2: row 73712621: reading CSV record: unexpected EOF
		    import failed
		    main.registerImportTPCH.func1.2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/import.go:160
		    main.(*monitor).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2168
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
```

<details><summary>Repro</summary><p>


Artifacts: [/import/tpch/nodes=8](https://teamcity.cockroachdb.com/viewLog.html?buildId=1676715&tab=artifacts#/import/tpch/nodes=8)
```
make stressrace TESTS=import/tpch/nodes=8 PKG=./pkg/roachtest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).import/tpch/nodes=8 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1676733&tab=buildLog) on [release-19.2@f3bcbd305602c89d52cb7ea4cb114f362c5aee23](https://github.com/cockroachdb/cockroach/commits/f3bcbd305602c89d52cb7ea4cb114f362c5aee23):

```
The test failed on branch=release-19.2, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20200107-1676733/import/tpch/nodes=8/run_1
	cluster.go:2189,import.go:164,test_runner.go:716: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2180
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2188
		    main.registerImportTPCH.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/import.go:164
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:716
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2233
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - pq: gs://cockroach-fixtures/tpch-csv/sf-100/lineitem.tbl.4: row 70178111: reading CSV record: unexpected EOF
		    import failed
		    main.registerImportTPCH.func1.2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/import.go:160
		    main.(*monitor).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2168
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
```

<details><summary>Repro</summary><p>


Artifacts: [/import/tpch/nodes=8](https://teamcity.cockroachdb.com/viewLog.html?buildId=1676733&tab=artifacts#/import/tpch/nodes=8)
```
make stressrace TESTS=import/tpch/nodes=8 PKG=./pkg/roachtest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).import/tpch/nodes=8 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1702345&tab=buildLog) on [master@a5b42be47cf47206d2343f40bbaf445f387f3890](https://github.com/cockroachdb/cockroach/commits/a5b42be47cf47206d2343f40bbaf445f387f3890):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20200121-1702345/import/tpch/nodes=8/run_1
	cluster.go:2201,import.go:164,test_runner.go:716: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2192
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2200
		    main.registerImportTPCH.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/import.go:164
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:716
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2245
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - pq: gs://cockroach-fixtures/tpch-csv/sf-100/lineitem.tbl.3: row 22142576: reading CSV record: unexpected EOF
		    import failed
		    main.registerImportTPCH.func1.2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/import.go:160
		    main.(*monitor).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2180
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357

```

<details><summary>Repro</summary><p>


Artifacts: [/import/tpch/nodes=8](https://teamcity.cockroachdb.com/viewLog.html?buildId=1702345&tab=artifacts#/import/tpch/nodes=8)
[roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Aimport%2Ftpch%2Fnodes%3D8.%2A&sort=title&restgroup=false&display=lastcommented+project)

<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).import/tpch/nodes=8 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1745452&tab=buildLog) on [master@de7a7bb804716835579e713ab7264f8af168a14f](https://github.com/cockroachdb/cockroach/commits/de7a7bb804716835579e713ab7264f8af168a14f):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/import/tpch/nodes=8/run_1
	cluster.go:2368,import.go:167,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerImportTPCH.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/import.go:167
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - pq: gs://cockroach-fixtures/tpch-csv/sf-100/lineitem.tbl.7: row 66345562: reading CSV record: unexpected EOF
		    import failed
		    main.registerImportTPCH.func1.2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/import.go:163
		    main.(*monitor).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2344
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357

```

<details><summary>More</summary><p>


Artifacts: [/import/tpch/nodes=8](https://teamcity.cockroachdb.com/viewLog.html?buildId=1745452&tab=artifacts#/import/tpch/nodes=8)
Related:
- #44593 roachtest: import/tpch/nodes=8 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Aimport%2Ftpch%2Fnodes%3D8.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).import/tpch/nodes=8 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1757483&tab=buildLog) on [master@fde991a0f5aba7ffd66112295996ad26f3725036](https://github.com/cockroachdb/cockroach/commits/fde991a0f5aba7ffd66112295996ad26f3725036):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/import/tpch/nodes=8/run_1
	cluster.go:2368,import.go:167,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerImportTPCH.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/import.go:167
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - pq: gs://cockroach-fixtures/tpch-csv/sf-100/lineitem.tbl.1: row 37124299: reading CSV record: unexpected EOF
		    import failed
		    main.registerImportTPCH.func1.2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/import.go:163
		    main.(*monitor).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2344
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357

```

<details><summary>More</summary><p>


Artifacts: [/import/tpch/nodes=8](https://teamcity.cockroachdb.com/viewLog.html?buildId=1757483&tab=artifacts#/import/tpch/nodes=8)
Related:
- #45074 roachtest: import/tpch/nodes=8 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-44941](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-44941) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44593 roachtest: import/tpch/nodes=8 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Aimport%2Ftpch%2Fnodes%3D8.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).import/tpch/nodes=8 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1786843&tab=buildLog) on [master@c9b189b3a2171c8da78c75fe80dd02933e18a5e5](https://github.com/cockroachdb/cockroach/commits/c9b189b3a2171c8da78c75fe80dd02933e18a5e5):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20200305-1786843/import/tpch/nodes=8/run_1
	cluster.go:2368,import.go:167,test_runner.go:741: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerImportTPCH.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/import.go:167
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:741
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - pq: gs://cockroach-fixtures/tpch-csv/sf-100/lineitem.tbl.6: unexpected EOF
		    import failed
		    main.registerImportTPCH.func1.2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/import.go:163
		    main.(*monitor).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2344
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357

```

<details><summary>More</summary><p>


Artifacts: [/import/tpch/nodes=8](https://teamcity.cockroachdb.com/viewLog.html?buildId=1786843&tab=artifacts#/import/tpch/nodes=8)
Related:
- #45074 roachtest: import/tpch/nodes=8 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-44941](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-44941) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44593 roachtest: import/tpch/nodes=8 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Aimport%2Ftpch%2Fnodes%3D8.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(roachtest).import/tpch/nodes=8 failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1804460&tab=buildLog) on [master@793a9200c16693aff32aa6a4dd9d8bbcbddb30aa](https://github.com/cockroachdb/cockroach/commits/793a9200c16693aff32aa6a4dd9d8bbcbddb30aa):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20200312-1804460/import/tpch/nodes=8/run_1
	cluster.go:2368,import.go:167,test_runner.go:747: error with attached stack trace:
		    main.(*monitor).WaitE
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2356
		    main.(*monitor).Wait
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2364
		    main.registerImportTPCH.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/import.go:167
		    main.(*testRunner).runTest.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:747
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor failure:
		  - error with attached stack trace:
		    main.(*monitor).wait.func2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2412
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357
		  - monitor task failed:
		  - pq: communication error: rpc error: code = Canceled desc = context canceled
		    import failed
		    main.registerImportTPCH.func1.2
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/import.go:163
		    main.(*monitor).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2344
		    github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
		    	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
		    runtime.goexit
		    	/usr/local/go/src/runtime/asm_amd64.s:1357

```

<details><summary>More</summary><p>


Artifacts: [/import/tpch/nodes=8](https://teamcity.cockroachdb.com/viewLog.html?buildId=1804460&tab=artifacts#/import/tpch/nodes=8)
Related:
- #45074 roachtest: import/tpch/nodes=8 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-44941](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-44941) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #44593 roachtest: import/tpch/nodes=8 failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Aimport%2Ftpch%2Fnodes%3D8.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/storage/cloud/external_storage.go
+++ b/pkg/storage/cloud/external_storage.go
@@ -26,6 +26,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/settings"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/util/retry"
+	"github.com/cockroachdb/cockroach/pkg/util/sysutil"
 	"github.com/cockroachdb/errors"
 )
 
@@ -377,3 +378,26 @@ func delayedRetry(ctx context.Context, fn func() error) error {
 		return err
 	})
 }
+
+// isResumableHTTPError returns true if we can
+// resume download after receiving an error 'err'.
+// We can attempt to resume download if the error is ErrUnexpectedEOF.
+// In particular, we should not worry about a case when error is io.EOF.
+// The reason for this is two-fold:
+//   1. The underlying http library converts io.EOF to io.ErrUnexpectedEOF
+//   if the number of bytes transferred is less than the number of
+//   bytes advertised in the Content-Length header.  So if we see
+//   io.ErrUnexpectedEOF we can simply request the next range.
+//   2. If the server did *not* advertise Content-Length, then
+//   there is really nothing we can do: http standard says that
+//   the stream ends when the server terminates connection.
+// In addition, we treat connection reset by peer errors (which can
+// happen if we didn't read from the connection too long due to e.g. load),
+// the same as unexpected eof errors.
+func isResumableHTTPError(err error) bool {
+	return errors.Is(err, io.ErrUnexpectedEOF) || sysutil.IsErrConnectionReset(err)
+}
+
+// Maximum number of times we can attempt to retry reading from external storage,
+// without making any progress.
+const maxNoProgressReads = 3

--- a/pkg/storage/cloud/gcs_storage.go
+++ b/pkg/storage/cloud/gcs_storage.go
@@ -24,6 +24,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/util/contextutil"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/retry"
 	"github.com/cockroachdb/errors"
 	"golang.org/x/oauth2/google"
@@ -160,9 +161,63 @@ func (g *gcsStorage) WriteFile(ctx context.Context, basename string, content io.
 	return errors.Wrap(err, "write to google cloud")
 }
 
+// resumingGoogleStorageReader is a GS reader which retries
+// reads in case of a transient connection reset by peer error.
+type resumingGoogleStorageReader struct {
+	ctx    context.Context   // Reader context
+	bucket *gcs.BucketHandle // Bucket to read the data from
+	object string            // Object to read
+	data   io.ReadCloser     // Currently opened object data stream
+	pos    int64             // How much data was received so far
+}
+
+var _ io.ReadCloser = &resumingGoogleStorageReader{}
+
+func (r *resumingGoogleStorageReader) Read(p []byte) (n int, err error) {
+	for retries := 0; n == 0 && err == nil; retries++ {
+		n, err = r.data.Read(p)
+		r.pos += int64(n)
+
+		if err != nil && !errors.IsAny(err, io.EOF, io.ErrUnexpectedEOF) {
+			log.Errorf(r.ctx, "GCS:Read err: %s", err)
+		}
+
+		if isResumableHTTPError(err) {
+			log.Errorf(r.ctx, "GCS:Retry: error %s", err)
+			if retries > maxNoProgressReads {
+				err = errors.Wrap(err, "multiple Read calls return no data")
+				return
+			}
+			err = r.doRead()
+		}
+	}
+
+	return
+}
+
+func (r *resumingGoogleStorageReader) Close() error {
+	if r.data != nil {
+		return r.data.Close()
+	}
+	return nil
+}
+
+func (r *resumingGoogleStorageReader) doRead() error {
+	if r.data != nil {
+		if err := r.data.Close(); err != nil {
+			return err
+		}
+	}
+
+	return delayedRetry(r.ctx, func() error {
+		var readErr error
+		r.data, readErr = r.bucket.Object(r.object).NewRangeReader(r.ctx, r.pos, -1)
+		return readErr
+	})
+}
+
 func (g *gcsStorage) ReadFile(ctx context.Context, basename string) (io.ReadCloser, error) {
 	// https://github.com/cockroachdb/cockroach/issues/23859
-
 	var rc io.ReadCloser
 	err := delayedRetry(ctx, func() error {
 		var readErr error

--- a/pkg/storage/cloud/http_storage.go
+++ b/pkg/storage/cloud/http_storage.go
@@ -30,7 +30,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/contextutil"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/retry"
-	"github.com/cockroachdb/cockroach/pkg/util/sysutil"
 	"github.com/cockroachdb/errors"
 )
 
@@ -227,27 +226,6 @@ func (r *resumingHTTPReader) requestNextRange() (err error) {
 	}
 	return
 }
-
-// isResumableHTTPError returns true if we can
-// resume download after receiving an error 'err'.
-// We can attempt to resume download if the error is ErrUnexpectedEOF.
-// In particular, we should not worry about a case when error is io.EOF.
-// The reason for this is two-fold:
-//   1. The underlying http library converts io.EOF to io.ErrUnexpectedEOF
-//   if the number of bytes transferred is less than the number of
-//   bytes advertised in the Content-Length header.  So if we see
-//   io.ErrUnexpectedEOF we can simply request the next range.
-//   2. If the server did *not* advertise Content-Length, then
-//   there is really nothing we can do: http standard says that
-//   the stream ends when the server terminates connection.
-// In addition, we treat connection reset by peer errors (which can
-// happen if we didn't read from the connection too long due to e.g. load),
-// the same as unexpected eof errors.
-func isResumableHTTPError(err error) bool {
-	return errors.Is(err, io.ErrUnexpectedEOF) || sysutil.IsErrConnectionReset(err)
-}
-
-const maxNoProgressReads = 3
 
 // Read implements io.Reader interface to read the data from the underlying
 // http stream, issuing additional requests in case download is interrupted.

